compiler/rustc_target/src/spec/targets/nvptx64_nvidia_cuda.rs

```diff
@@ -46,6 +47,9 @@ pub fn target() -> Target {
             // The LLVM backend does not support stack canaries for this target
             supports_stack_protector: false,
 
+            // Support using `self-contained` linkers like the llvm-bitcode-linker
+            link_self_contained: LinkSelfContainedDefault::True,
```

Comment:
`-Clink-self-contained=+linker` is supposed to ensure that the directory inside Rust distribution in which `llvm-bitcode-linker` binary resides is in `PATH` when rustc is searching for the linker by name.
`-Clink-self-contained=-linker`, on the other hand, is supposed to ensure that the directory inside Rust distribution in which `llvm-bitcode-linker` binary resides is NOT in `PATH` when rustc is searching for the linker by name.

Does it actually work like that now?
I have some suspicion that the directory is always in `PATH`.

Reply:
It looks you are completely correct. The test still passes with `-Clink-self-contained=-linker`. Do you know if this is a problem with how the bootstrap is implemented for `llvm-bitcode-linker` or the `self-contained` machinery is a whole?

Reply:
What is the location of `llvm-bitcode-linker` in the distribution?

All shipped linkers should eventually be placed into the `.../self-contained/linker` subdirectory, which will be added/removed to/from `PATH`, but that's not the case right now because `-Clink-self-contained=+/-linker` is not fully implemented.

Reply:
The current situation is probably fine as is for this PR.
But it does make this target spec flag and `-Clink-self-contained` options in tests unnecessary.

Reply:
> What is the location of llvm-bitcode-linker in the distribution?

Right now it will be put in `bin`.

 Is the correct `self-contained` folder actually `lib/rustlib/<HOST>/lib/self-contained` or is that a different `self-contained` dir?
 
 > But it does make this target spec flag and -Clink-self-contained options in tests unnecessary.
 
I interpret that as you want me to remove them? I guess it will be difficult to make sure they're there for new tests as well so we will need to fix them when this `+linker`/`-linker` works anyway.

Reply:
I removed the  `-Clink-self-contained` statements now

Reply:
> Is the correct `self-contained` folder actually `lib/rustlib/<HOST>/lib/self-contained` or is that a different `self-contained` dir?

I see two self-contained directories in my toolchain - `lib/rustlib/<target>/bin/self-contained` and `lib/rustlib/<target>/lib/self-contained`.
Linkers are supposed to be in the `bin` one, including LLD.

>Is `lld` also supposed to work like this or will `llvm-bitcode-linker` be the first linker to actually work like this? 

Some copies of lld can be found in `lib/rustlib/<target>/bin/gcc-ld` subdirectory, and they indeed work like this, at least when LLD is called through a C/C++ compiler - the `gcc-ld` subdirectory is either considered or not depending on `-Clink-self-contained=+/-linker`.
LLD will also need to be moved to `lib/rustlib/<target>/bin/self-contained/linker` eventually.